api/v1alpha1/credential_types.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type CredentialState string
+
+const (
+	CredentialReady     CredentialState = "Ready"
+	CredentialNotFound  CredentialState = "Cluster Identity not found"
+	CredentialWrongType CredentialState = "Mismatched type"
+)
+
+type CredentialReference struct {
+	// +optional
+	Name string `json:"name,omitempty"`
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+}
+
+// CredentialSpec defines the desired state of Credential
+type CredentialSpec struct {
+	IdentityRef *corev1.ObjectReference `json:"identityRef"`
+	// +optional
+	Description string `json:"description,omitempty"`
+}
+
+// CredentialStatus defines the observed state of Credential
+type CredentialStatus struct {
+	State CredentialState `json:"state,omitempty"`
```

Comment:
Let's use conditions instead (but not necessary to change it now)

Reply:
Ok, I'll leave the state field for now and create a separate task for that